src/deepsparse/v2/text_generation/nl_engine_operator.py

```diff
@@ -86,43 +150,80 @@ def __init__(
 
         kwargs["engine_kwargs"] = engine_kwargs
         kwargs["model_path"] = onnx_file_path
+
         super().__init__(**kwargs)
 
-        self.input_ids_length = input_ids_length
+    def override_model(self, model_path: Union[str, Path], batch_size: int):
+        """
+        Override the model based on the provided batch_size, sequence_length,
+        and input_ids_length.
+        """
+        (
+            onnx_file_path,
+            output_indices_to_be_cached,
+            kv_cache_data_type,
+        ) = overwrite_onnx_model_inputs_for_kv_cache_models(
+            onnx_file_path=model_path,
+            batch_size=batch_size,
+            sequence_length=self.sequence_length,
+            input_ids_length=self.input_ids_length,
+        )
+        return onnx_file_path, output_indices_to_be_cached, kv_cache_data_type
 
-    def run(self, inp: NlEngineInput, **kwargs) -> Any:
+    def run(self, inp: NLEngineInputs, **kwargs) -> NLEngineOutputs:
```

Comment:
The run method got super bloated. Can we possibility refactor it, e.g. creating helper methods for better readibility?

Reply:
I can refactor once the case for internal kv cache is also added.